src/vmm/src/builder.rs

```diff
@@ -556,7 +565,22 @@ pub fn build_microvm_from_snapshot(
     .map_err(Error::SeccompFilters)
     .map_err(StartMicrovmError::Internal)?;
 
-    Ok(vmm)
+    {
+        let vmm = vmm_reference.lock().expect("Poisoned lock");
+
+        vm_resources.set_vm_config(&VmConfig {
+            vcpu_count: Some(vmm.vcpus_handles.len() as u8),
```

Comment:
There is already a `vcpu_count` variable in scope that we may use. Same for `track_dirty_pages` and `guest_memory`